app/helpers/patient_query_helper.rb

```diff
@@ -281,18 +281,52 @@ def advanced_filter(patients, filters, tz_offset)
         patients = advanced_filter_relative_date(patients, :symptom_onset, filter, tz_diff, :date)
       when 'continous-exposure'
         patients = patients.where(continuous_exposure: filter[:value].present? ? true : [nil, false])
+      when 'close-contact-with-known-case'
+        case filter[:additionalFilterOption]
+        when 'Exact Match'
+          patients = if filter[:value].blank?
+                       patients.where(contact_of_known_case_id: [nil, ''])
+                     else
+                       patients.where('patients.contact_of_known_case_id = ?', filter[:value])
+                     end
+        when 'Contains'
+          patients = if filter[:value].blank?
+                       patients.where(contact_of_known_case_id: [nil, ''])
+                     else
+                       patients.where('patients.contact_of_known_case_id like ?', "%#{filter[:value]}%")
+                     end
+        when 'Multiple'
+          if filter[:value].blank?
+            patients = patients.where(contact_of_known_case_id: [nil, ''])
+          else
+            value_array = filter[:value].split(/\s*,\s*/)
+            filtered_patient_ids = []
+            value_array.each do |value|
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id = ?', value).pluck(:id)
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id like ?', "#{value},%").pluck(:id)
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id like ?', "%,#{value},%").pluck(:id)
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id like ?', "%, #{value},%").pluck(:id)
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id like ?', "%,#{value}").pluck(:id)
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id like ?', "%, #{value}").pluck(:id)
```

Comment:
I'm confused why you are making 6 queries per value - this is really inefficient. 
I see the cases (where "..." means any other characters):
1. exact match
2. value,..
3.  ...,value,...
4. ..., value,...
5. ...,value
6. ..., value

Why do you need to specifically check each of these? It seems like "Contains" option handles the main need, but we're adding this entire "Multiples" option so that we can filter out whitespace surrounding the commas? I might just be missing an aspect of the user requests/requirements here but I'm confused why the Multiple option is even necessary.

If the above isn't do-able for some reason I'm not thinking of, I'd just use `REGEXP` instead of `LIKE` here if we absolutely need to handle these possible cases, and you'd just need to come up with the regex that works. That would bring it down to 1 query total rather than 6 per value + 1 final query as it is now.

Reply:
Multiple is the only filter option that takes into account that there could be multiple values passed in through the text input.  Now that I'm thinking about it though we could scrap the "exact match" and "contains" options because I think the multiple option would suffice for what the users are looking for.  It would work if you enter one value or multiple.  Let me know what you think on this.

In regards to it being inefficient, I'm aware and there isn't really any other way around it.  The goal here is to prevent the case of returning a monitoree with close contact of known case of `12345678` and you search `45`.  I was under the impression from the PO team that we would want an exact match, hence why there are all the additional cases and checks for commas.  I'm open to other ideas but to my understanding this is the exact functionality that they want

Reply:
Really the core problem is that the way we handle Case ID is not great and it's resulting in unstandardized data input that causes design problems like this. Regardless, I think we can find a better solution.

> Now that I'm thinking about it though we could scrap the "exact match" and "contains" options because I think the multiple option would suffice for what the users are looking for. It would work if you enter one value or multiple. Let me know what you think on this.

Well the Multiple option doesn't really capture the Contains option since it's more strict, so I wouldn't merge those two. You're right it does already capture the Exact case. I'd prefer to keep it separate for UX reasons I think, but willing to have discussion with PO team if you'd like.

> In regards to it being inefficient, I'm aware and there isn't really any other way around it

I'm not nearly as certain that is true - see REGEXP option I mentioned above. I think you can write a regex that captures the cases outlined above for the Multiple case while also not being as flexible as the Contains option (to prevent the example case with `12345678` and `45` that you mentioned above). See [REGEXP](https://dev.mysql.com/doc/refman/8.0/en/regexp.html#operator_regexp) in MySQL.

Reply:
updated